### ISSUE TITLE
Hexadecimal alignment format changed from v7 to v8?

### ISSUE THREAD

**Issue by chris-t-w** [fmtlib/fmt#2900]
Alignment of the hexadecimal prefix seems to have changed between version 7.1.3 and 8.0.1:

```c++
fmt::print("{:0>#016x}\n", 0xaaaabbbb);
```

Results:

* 7.1.3 - `0x000000aaaabbbb` - https://godbolt.org/z/sYc95WGh3
* 8.0.1 - `0000000xaaaabbbb` - https://godbolt.org/z/T89zcjdGa

Was this intentional?

**Comment by vitaut** [comment]
Yes, this has been changed for consistency with `std::format` spec:

> If the 0 character and an align option both appear, the 0 character is ignored.

To get the old formatting you should use the `0` spec alone (which works in both versions and is not ambiguous):

```c++
fmt::print("{:#018x}\n", 0xaaaabbbb);
```

https://godbolt.org/z/Pc5EqMMaT
